cabot/cabotapp/alert.py

```diff
@@ -23,17 +24,57 @@ def send_alert(self, service, users, duty_officers):
         """
         return True
 
+    def send_test_alert(self, user):
+        """
+        Send a test alert when the user requests it (to make sure config is valid).
+
+        The default implementation creates a fake Service and StatusCheck and calls the normal send_alert().
+        Note that the service/check may not be fully configured (e.g. invalid primary key, no HipChat room ID, ...).
+        :param user: django user
+        :return: nothing, raise exceptions on error
+        """
+        # local import because models.py depends on alert.py :\
+        from cabot.cabotapp.models import CheckGroupMixin
+
+        check_mock = Mock()
+        check_mock.configure_mock(id=0, pk=0, name='Alert Testing Check', active=True)
+
+        service_mock = Mock()
+        service_mock.configure_mock(id=0, pk=0, name='Alert Testing Service', alerts_enabled=True,
+                                    CRITICAL_STATUS=CheckGroupMixin.CRITICAL_STATUS,  # hipchat plugin uses this
```

Comment:
How hard would it be to move these into a defs.py file, referred to by both alert.py and models.py? No reason they have to be attached to the model, would be a nice cleanup (and then we don't need the local import)

Reply:
By these, you mean `CRITICAL_STATUS` etc. right? That's a great idea, hadn't thought of that - will do.